### PR TITLE
implemented product not found exception

### PR DESCRIPTION
**Pull request by PauliusKibilda**

### CODE REVIEW

File: FoodWaste/Controllers/ProductsController.cs [hunk 1]
```diff
@@ -51,6 +53,11 @@ public async Task<IActionResult> Index(string sortOrder, string searchString)
                 _ => products.OrderBy(p => p.Name).ToList(),
             };
             return View(products);
+            }
```
Comment (MindaugasJusis) [inline FoodWaste/Controllers/ProductsController.cs:56]: reiketu sutvarkyti formatavima

File: FoodWaste/Controllers/ProductsController.cs [hunk 2]
```diff
@@ -240,17 +281,35 @@ public async Task<IActionResult> DeleteConfirmed(int id)
...
+            try
+            {
+                var result = await DataBaseOperations.GetProduct();
+                var product = result.FirstOrDefault(m => m.Id == id);
+                await DataBaseOperations.DeleteProduct(id);
+                return RedirectToAction(nameof(Index));
+            }
+            catch (Exception ex)
+            {
+                return LocalRedirect("/Products/NotFound");
```
Comment (VaidasCib) [inline FoodWaste/Controllers/ProductsController.cs:293]: create a canstant called NotFoundPage

File: FoodWaste/Controllers/ProductsController.cs [hunk 3]
```diff
@@ -222,14 +256,21 @@ public async Task<IActionResult> Edit(int? id)
...
+                {
+                    return NotFound();
+                }
 
-            return View(product);
+                return View(product);
+            }
+            catch (Exception ex)
+            {
+                return LocalRedirect("/Products/NotFound");
```
Comment (VaidasCib) [inline FoodWaste/Controllers/ProductsController.cs:272]: create a canstant called NotFoundPage

File: FoodWaste/Controllers/ProductsController.cs [hunk 4]
```diff
@@ -167,14 +194,21 @@ public async Task<IActionResult> Edit(int? id)
...
+                if (product == null)
+                {
+                    return NotFound();
+                }
+                return View(product);
+            }
+            catch (Exception ex)
             {
-                return NotFound();
+                return LocalRedirect("/Products/NotFound");
```
Comment (VaidasCib) [inline FoodWaste/Controllers/ProductsController.cs:210]: create a canstant called NotFoundPage

File: FoodWaste/Controllers/ProductsController.cs [hunk 5]
```diff
@@ -152,8 +172,15 @@ public async Task<IActionResult> Create()
...
-                await DataBaseOperations.PostProduct(product);
-                return RedirectToAction(nameof(Index));
+                try
+                {
+                    await DataBaseOperations.PostProduct(product);
+                    return RedirectToAction(nameof(Index));
+                }
+                catch (Exception ex)
+                {
+                    return LocalRedirect("/Products/NotFound");
```
Comment (VaidasCib) [inline FoodWaste/Controllers/ProductsController.cs:182]: create a canstant called NotFoundPage

File: FoodWaste/Controllers/ProductsController.cs [hunk 6]
```diff
@@ -113,19 +126,26 @@ public async Task<IActionResult> Details(int? id)
...
+                {
+                    return NotFound();
+                }
 
-            return View(product);
+                return View(product);
+            }
+            catch (Exception ex)
+            {
+                return LocalRedirect("/Products/NotFound");
```
Comment (VaidasCib) [inline FoodWaste/Controllers/ProductsController.cs:147]: create a canstant called NotFoundPage

File: FoodWaste/Controllers/ProductsController.cs [hunk 7]
```diff
@@ -64,39 +70,46 @@ public async Task<IActionResult> Reserve(int? id)
...
+                        }
+                        else
+                        {
+                            throw;
+                        }
+                    }
+                }
+                catch (Exception ex)
+                {
+                    LocalRedirect("/Products/NotFound");
```
Comment (VaidasCib) [inline FoodWaste/Controllers/ProductsController.cs:112]: create a canstant called NotFoundPage

File: FoodWaste/Controllers/ProductsController.cs [hunk 8]
```diff
@@ -33,24 +33,30 @@ public async Task<IActionResult> Index(string sortOrder, string searchString)
...
             {
-                "Name_desc" => products.OrderByDescending(p => p.Name).ToList(),
-                "Date" => products.OrderBy(p => p.ExpiryDate).ToList(),
-                "Date_desc" => products.OrderByDescending(p => p.ExpiryDate).ToList(),
-                "State" => products.OrderBy(p => p.State).ToList(),
-                "State_desc" => products.OrderByDescending(p => p.State).ToList(),
-                _ => products.OrderBy(p => p.Name).ToList(),
-            };
-            return View(products);
+                return LocalRedirect("/Products/NotFound");
```
Comment (VaidasCib) [inline FoodWaste/Controllers/ProductsController.cs:58]: create a canstant called NotFoundPage

**Comment by VaidasCib** [comment]
looks good now

**Pull request merged**